src/status_im2/contexts/chat/messages/link_preview/view.cljs

```diff
@@ -0,0 +1,160 @@
+  (:require
+   [clojure.string :as string]
+   [i18n.i18n :as i18n]
+   [reagent.core :as reagent]
+   [react-native.core :as rn]
+   [status-im.constants :as constants]
+   [status-im2.contexts.chat.messages.link-preview.events]
+   [status-im2.contexts.chat.messages.link-preview.style :as style]
+   [quo2.core :as quo]
+   [utils.re-frame :as rf]
+   [utils.security.core :as security]))
+
+(defn link-belongs-to-domain
+  [link domain]
+  (string/starts-with? link (str "https://" domain)))
+
+(defn community-id-from-link
+  [link]
+  (nth (re-find constants/regx-community-universal-link link) 4))
+
+(defn domain-info-if-whitelisted
+  [link whitelist]
+  (->> whitelist
+       (filter #(link-belongs-to-domain link (:address %)))
+       first))
+
+(defn link-extended-info
+  [link whitelist enabled-list]
+  (if-not (community-id-from-link link)
+    (let [domain-info (domain-info-if-whitelisted link whitelist)]
+      {:whitelisted? (boolean domain-info)
+       :enabled?     (contains? enabled-list (:title domain-info))
+       :link         link})
+    {:whitelisted? true
+     :enabled?     true
+     :link         link
+     :community?   true}))
+
+(defn previewable-link
+  [links whitelist enabled-list]
+  (->> links
+       (map #(link-extended-info % whitelist enabled-list))
+       (filter #(:whitelisted? %))
+       (first)))
+
+(defn is-gif?
+  [url]
+  (and url (string/ends-with? url ".gif")))
+
+(defn community-preview
+  [{:keys [name members description verified] :as community}]
+  (let [members-count (count members)]
+    [rn/view (style/wrapper)
+     [rn/view (style/title-wrapper)
+      [rn/image {:style (style/title-site-image)}]
+      [rn/text {:style (style/title-text)}
+       name]]
+     [rn/text {:style (style/extra-text)}
+      (if verified
+        (i18n/label :t/verified-community)
+        (i18n/label :t/community))]
+     [rn/text {:style (style/main-text)}
+      description]
+     [rn/text {:style (style/extra-text)}
+      (i18n/label-pluralize members-count :t/community-members {:count members-count})]
+     [rn/view (style/separator)]
+     [quo/button
+      {:type     :grey
+       :on-press #(rf/dispatch [:navigate-to :community
+                                {:from-chat    true
+                                 :community-id (:id community)}])}
+      (i18n/label :view)]]))
+
+(defn community-preview-loader
+  [community-link]
+  (let [cache (rf/sub [:link-preview/cache])]
+    (reagent/create-class
+     {:component-did-mount
+      (fn []
+        (let [community    (get cache community-link)
+              community-id (community-id-from-link community-link)]
+          (when-not community
+            (rf/dispatch [:chat.ui/resolve-community-info community-id]))))
+      :reagent-render
+      (fn []
+        (when-let [community (get cache community-link)]
+          [community-preview community]))})))
+
+(defn link-preview-loader
+  [link]
+  (let [cache (rf/sub [:link-preview/cache])]
```

Comment:
why we are subscribing to the complete cache here? Can we pass the link to the subscription and only subscribe to that?

Reply:
Fixed!